Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationFailureRepository.cs

```diff
@@ -0,0 +1,234 @@
+
+      var failures = failureContainer.GetValidationFailuresForDataRow().Where(f => !f.Handled);
+
+      if (markAsHandled)
+        failures = failures.ApplySideEffect(f => f.MarkAsHandled());
+
+      return failures.Select(f => f.Failure).ToList();
+    }
+
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledValidationFailuresForDataRowAndContainingDataCells (IBusinessObject rowObject, bool markAsHandled)
+    {
+      ArgumentUtility.CheckNotNull(nameof(rowObject), rowObject);
+
+      var failureContainer = _rowValidationFailures.GetValueOrDefault(rowObject);
+      if (failureContainer == null)
+        return Array.Empty<BusinessObjectValidationFailure>();
+
+      var failures = failureContainer.GetValidationFailuresForDataRow().Where(f => !f.Handled);
+      failures = failures.Concat(failureContainer.GetAllCellFailuresForRow().Where(f => !f.Handled));
+
+      if (markAsHandled)
+        failures = failures.ApplySideEffect(f => f.MarkAsHandled());
+
+      return failures.Select(f => f.Failure).ToList();
+
+    }
+
+    public IReadOnlyCollection<BusinessObjectValidationFailure> GetUnhandledValidationFailuresForDataCell (IBusinessObject rowObject, IBocColumnDefinitionWithValidationSupport columnDefinition, bool markAsHandled)
+    {
+      ArgumentUtility.CheckNotNull(nameof(rowObject), rowObject);
+      ArgumentUtility.CheckNotNull(nameof(columnDefinition), columnDefinition);
+
+      var failureContainer = _rowValidationFailures.GetValueOrDefault(rowObject);
+      if (failureContainer == null)
+        return Array.Empty<BusinessObjectValidationFailure>();
+
+      var cellFailures = failureContainer.GetValidationFailuresForDataCell(columnDefinition).Where(f => !f.Handled);
+
+      if (markAsHandled)
+        cellFailures = cellFailures.ApplySideEffect(f => f.MarkAsHandled());
+
+      return cellFailures.Select(f => f.Failure).ToList();
+    }
+
+    public bool HasValidationFailuresForDataRow (IBusinessObject rowObject)
+    {
+      ArgumentUtility.CheckNotNull(nameof(rowObject), rowObject);
+
+      return _rowValidationFailures.GetValueOrDefault(rowObject)?.HasValidationFailures() ?? false;
+    }
+
+    public void ClearAllValidationFailures ()
+    {
+      _rowValidationFailures.Clear();
+      _listValidationFailures.Clear();
+    }
+
+    private RowFailureContainer GetOrCreateRowFailureContainer (IBusinessObject rowObject)
+    {
+      if (_rowValidationFailures.TryGetValue(rowObject, out var existingFailureContainer))
+        return existingFailureContainer;
+
+      var newFailureContainer = new RowFailureContainer();
+      _rowValidationFailures.Add(rowObject, newFailureContainer);
+
+      return newFailureContainer;
+    }
+
+    private IEnumerable<HandleableFailure> GetUnhandledValidationFailuresForBocListInternal ()
+    {
+
+      var failures= _listValidationFailures.Where(e => !e.Handled);
+      return failures;
+    }
```

Comment:
given the rest of the code, we can now inline this. There's is a lot of similiarity with _rowValidationFailures and those also have the handled-filter inline. It will make the code actually more compact / easier to read because we are able to unify the handled-calls.